l2-contracts/contracts/bridge/L2SharedBridge.sol

```diff
@@ -39,169 +40,160 @@ contract L2SharedBridge is IL2SharedBridge, Initializable {
-        return abi.encodePacked(IL1ERC20Bridge.finalizeWithdrawal.selector, _to, _l1Token, _amount);
+        // solhint-disable-next-line func-named-parameters
+        return abi.encodePacked(IL1ERC20Bridge.finalizeWithdrawal.selector, _assetId, _bridgeMintData);
     }
 
-    /// @return Address of an L2 token counterpart
-    function l2TokenAddress(address _l1Token) public view override returns (address) {
-        bytes32 constructorInputHash = keccak256(abi.encode(address(l2TokenBeacon), ""));
-        bytes32 salt = _getCreate2Salt(_l1Token);
-        return
-            L2ContractHelper.computeCreate2Address(address(this), salt, l2TokenProxyBytecodeHash, constructorInputHash);
+    /// @dev Used to set the assedAddress for a given assetId.
+    /// @dev Will be used by ZK Gateway
+    function setAssetHandlerAddress(bytes32 _assetId, address _assetAddress) external onlyL1Bridge {
+        assetHandlerAddress[_assetId] = _assetAddress;
+        emit AssetHandlerRegistered(_assetId, _assetAddress);
     }
 
-    /// @dev Convert the L1 token address to the create2 salt of deployed L2 token
-    function _getCreate2Salt(address _l1Token) internal pure returns (bytes32 salt) {
-        salt = bytes32(uint256(uint160(_l1Token)));
+    /*//////////////////////////////////////////////////////////////
+                            LEGACY FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    function finalizeDeposit(
+        address _l1Sender,
+        address _l2Receiver,
+        address _l1Token,
+        uint256 _amount,
+        bytes calldata _data
+    ) external override {
+        // onlyBridge {
+        bytes32 assetId = keccak256(
+            abi.encode(L1_CHAIN_ID, NATIVE_TOKEN_VAULT_VIRTUAL_ADDRESS, bytes32(uint256(uint160(_l1Token))))
+        );
+        // solhint-disable-next-line func-named-parameters
+        bytes memory data = abi.encode(_l1Sender, _amount, _l2Receiver, _data, _l1Token);
+        finalizeDeposit(assetId, data);
     }
 
-    /// @dev Deploy the beacon proxy for the L2 token, while using ContractDeployer system contract.
-    /// @dev This function uses raw call to ContractDeployer to make sure that exactly `l2TokenProxyBytecodeHash` is used
-    /// for the code of the proxy.
-    function _deployBeaconProxy(bytes32 salt) internal returns (BeaconProxy proxy) {
-        (bool success, bytes memory returndata) = SystemContractsCaller.systemCallWithReturndata(
-            uint32(gasleft()),
-            DEPLOYER_SYSTEM_CONTRACT,
-            0,
-            abi.encodeCall(
-                IContractDeployer.create2,
-                (salt, l2TokenProxyBytecodeHash, abi.encode(address(l2TokenBeacon), ""))
+    function withdraw(address _l1Receiver, address _l2Token, uint256 _amount) external {
+        if (l1TokenAddress[_l2Token] == address(0)) {
+            getL1TokenAddress(_l2Token);
```

Comment:
this code does nothing (calls a view function and does not use the returned value)